src/storage/mutate/AddEdgesProcessor.cpp

```diff
@@ -134,40 +130,52 @@ AddEdgesProcessor::addEdges(PartitionID partId,
      *
      * Ultimately, kv(part1_src1_edgeType1_rank1_dst1 , v4) . It's just what I need.
      */
-    std::map<std::string, std::string> newEdges;
+    std::unordered_map<std::string, std::string> newEdges;
     std::for_each(edges.begin(), edges.end(),
                  [&newEdges](const std::map<std::string, std::string>::value_type& e)
                  { newEdges[e.first] = e.second; });
 
     for (auto& e : newEdges) {
         std::string val;
+        RowReaderWrapper oReader;
         RowReaderWrapper nReader;
         auto edgeType = NebulaKeyUtils::getEdgeType(spaceVidLen_, e.first);
+        auto srcId = NebulaKeyUtils::getSrcId(spaceVidLen_, e.first).str();
+        auto rank = NebulaKeyUtils::getRank(spaceVidLen_, e.first);
+        auto dstId = NebulaKeyUtils::getDstId(spaceVidLen_, e.first).str();
+
+        // Normal index
         for (auto& index : indexes_) {
             if (edgeType == index->get_schema_id().get_edge_type()) {
                 auto indexId = index->get_index_id();
-
                 /*
-                 * step 1 , Delete old version index if exists.
+                 * Step 1 , Delete old version normal index if exists.
                  */
                 if (val.empty()) {
-                    auto obsIdx = findObsoleteIndex(partId, e.first);
+                    auto obsIdx = findOldValue(partId, srcId, edgeType, rank, dstId);
                     if (obsIdx == folly::none) {
                         return folly::none;
                     }
                     val = std::move(obsIdx).value();
-                }
-
-                if (!val.empty()) {
-                    auto reader = RowReaderWrapper::getEdgePropReader(env_->schemaMan_,
+                    if (!val.empty()) {
+                        // put keyExist_, key exist
+                        keyExist_[std::make_tuple(srcId, edgeType, rank, dstId)] = true;
```

Comment:
Good question.
The logic here is to handle the normal index(from line 148)
If there is no normal index, it can't go here.

When both normal index and edge count index exist,
This can reduce the read data from kvstore.